fbw-a32nx/src/systems/fmgc/src/flightplanning/rpc/FlightPlanRpcServer.ts

```diff
@@ -22,19 +26,50 @@ export class FlightPlanRpcServer<P extends FlightPlanPerformanceData = FlightPla
     const sub = bus.getSubscriber<FlightPlanRemoteClientRpcEvents<P>>();
     this.pub = bus.getPublisher<FlightPlanServerRpcEvents>();
 
-    sub.on('flightPlanRemoteClient_rpcCommand').handle(([command, id, ...args]) => {
-      this.handleRpcCommand(command, id, ...args);
+    const gameStateSub = GameStateProvider.get().sub((state) => {
+      if (state === GameState.ingame) {
+        this.handleSendHeartbeat();
+        setInterval(() => this.handleSendHeartbeat(), 2_500);
```

Comment:
Do we need to clean up this `setInterval` when we destroy the RPC server?

Reply:
At the moment the server basically always exists, because it exists as long as the FMC does, and IIRC we don't have FMC failures yet. If you want, I can add a `destroy()` method

Reply:
I'm not too bothered about cleaning up to be honest if it's expected to exist for the entire session. It's fine to keep it as it is.